ion/reader.go

```diff
@@ -163,6 +163,17 @@ type Reader interface {
 
 	// IsInStruct indicates if the reader is currently positioned in a struct.
 	IsInStruct() bool
+
+	// FieldNameSymbol returns the field name associated with the current value as a SymbolToken.
+	// It returns an error if the current value has no field name.
+	FieldNameSymbol() (SymbolToken, error)
+
+	// SymbolValue returns the SymbolToken associated with the current value. It returns an
+	// error if the current value is not an Ion symbol.
+	SymbolValue() (SymbolToken, error)
+
+	// Clear clears the current value from the reader.
+	clear()
```

Comment:
I tried removing clear() and [TestReadNullLST](https://github.com/amzn/ion-go/blob/master/ion/binaryreader_test.go#L60-L69) test failed. This is because it is checking for [empty annotations.](https://github.com/amzn/ion-go/blob/master/ion/textreader_test.go#L820-L822)

I updated clear to be exportable

Reply:
I don't think we want `Clear()` to be part of the public API that users can implement. It feels like an implementation detail to me; no user would ever call `Clear()`.

In OO languages, you might have a `Writer` interface and an `AbstractWriter` base class to hold common implementation details like `Clear()`. I'm unsure what the best way to model that in Go would be.

Reply:
Removed `Clear()` from `Reader` interface. `Clear()` was being used in the beginning of [readLocalSymbolTable()](https://github.com/amzn/ion-go/pull/149/commits/851ab03c6a785fb196e6722d01ed0be28ed4ae9f#diff-c7a62f6b5becd9a9a65eb57ed6c40180L7-L10) and I move that logic to to text/binary readers just before they call `readLocalSymbolTable`